app/main.py

```diff
@@ -1,214 +1,56 @@
+from app.contestants.knight import Knight
+from app.equipment.armour import Armour
+from app.equipment.weapon import Weapon
+from app.equipment.potion import Potion
+from app.battle.battle_start import Battle
 
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
 
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
+def battle(knights_list: dict) -> dict:
 
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
+    list_of_knights = {}
 
-    # mordred
-    mordred = knightsConfig["mordred"]
+    # KNIGHTS INITIATION
+    for contestant, value in knights_list.items():
+        list_of_knights[contestant] = Knight(
+            name=value["name"],
+            power=value["power"],
+            hp=value["hp"],
+            armour=value["armour"],
+            weapon=value["weapon"],
+            potion=value["potion"]
+        )
 
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
+    # BATTLE PREPARATIONS:
+    for knight in list_of_knights.values():
+        armour = Armour(knight.armour)
+        armour.apply_armour(knight)
 
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
+        weapon = Weapon(knight.weapon["power"])
+        weapon.apply_weapon(knight)
 
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
+        if knight.potion is not None:
+            potion = Potion(knight.potion["effect"])
+            potion.apply_potion(knight)
 
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    print(list_of_knights["lancelot"].name)
```

Comment:
put it in the Knight class, so that it is called when a new knight is initialized

Reply:
when I put this part into the Knight class, the circular import occurs between modules knight/potion, weapon, armour. how to execute this part of code inside the knight module without circulat import error arising?